#' This function does nothing. It is intended to inherit is parameters'
#' documentation.
#' @param alleles one or more alleles, e.g. \code{DRB1_0101}.
#'   See \link{get_netmhc2pan_alleles} for a full list
#' @param binding_strength_threshold threshold of the MHC2 allele's
#'   binding strength. A low threshold, e.g. 0.05, selects for the top
#'   5 percent best binders
#' @param epitopeome_filename name of the file to store the epitopeome
#'   output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param fasta_filename the name of a FASTA file with protein sequences
#' @param netmhc2pan_filename name of the file to store the
#'   NetMHC2pan output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param tmhmm_filename name of the file to store the
#'   TMHMM output.
#'   By default, this filename is generated by \link{tempfile},
#'   and will be put in the \code{/tmp} folder,
#'   which is cleaned by the operating system
#' @param verbose set to TRUE for more output
#' @author Richel J.C. Bilderbeek
#' @note This is an internal function, so it should be marked with
#'   \code{@noRd}. This is not done, as this will disallow all
#'   functions to find the documentation parameters
default_params_doc <- function(
  alleles,
  binding_strength_threshold,
  epitopeome_filename,
  fasta_filename,
  netmhc2pan_filename,
  tmhmm_filename,
  verbose
) {
  # Nothing
}
